component/http/cache_builder.go

```diff
@@ -0,0 +1,116 @@
+package http
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/beatlabs/patron/cache"
+	errs "github.com/beatlabs/patron/errors"
+)
+
+// RouteCacheBuilder is the builder needed to build a cache for the corresponding route
+type RouteCacheBuilder struct {
```

Comment:
Since we have already a route builder we should use the same to integrate caching.
This will make the API even more developer friendly.

Reply:
The reason there is a separate builder, is that this cache builder has methods, that dont make sense if you have no cache enabled. So creating a method for example in the RouteBuilder : `WithCacheTimeout(timeout int)` does not make sense , if you have not called `WithCache(cache cache)` before ... So i tried to enforce it this way. We could also put the methods in the RouteBuilder and throw an error if the routeBuilder has not been defined to work with a cache (either the WithCache method has been called, or at the create call ... ) ? What do you think ?

Reply:
The reason I would like to put it to the RouteBuilder is that having another builder makes the API again complex as is was in the beginning.
Quick question: Timeout and instant as I understand it are cache specific and are not route HTTP specific. 
Would that not mean that the only thing you need is the `WithCache(cache)` in the route builder?

Reply:
yeah the cache related options specified by the builder are :
- timeinstant
- minAge
- maxfresh
and ...
- allowStaleResponses.
In that case , we would need to add these as arguments to the WithCache method ... and it would make it more difficult to apply default values (whereas the builder is a nice abstraction for that) ... if ai am not missing anything . 
But i am happy to go either way ...

Reply:
Actually the cache-related properties are TTL and time instant as I understand it. the rest are HTTP cache related.
Somebody will create a cache instance with TTL and time instant, which will be used in other routes also, and pass it into the route builder. correct?

Reply:
yes, good point! Although also the other properties are route specific, but still only relevant for routes with cache.

Reply:
removed the cacheRouteBuilder. And made the config thinner . What do you think ?